pkg/ingester/errors.go

```diff
@@ -263,14 +282,18 @@ func (e perUserSeriesLimitReachedError) Error() string {
 	)
 }
 
-// perUserSeriesLimitReachedError implements the ingesterError interface.
-func (e perUserSeriesLimitReachedError) errorType() ingesterErrorType {
-	return badData
+func (e perUserSeriesLimitReachedError) errorCause() mimirpb.ErrorCause {
+	return mimirpb.BAD_DATA
```

Comment:
I'm surprised the exceeding the user limit would result in bad data. 
But since in this PR you don't want to change the behavior this is out of scope of the PR

Reply:
@replay We can always add an additional cause and change it. Would it be more appropriate to mark it as `TENANT_LIMIT`?